Documentation/gettingstarted/cni-chaining-aws-cni.rst

```diff
@@ -0,0 +1,131 @@
+network devices as well as address allocation (IPAM) via ENI. After the initial
+networking is setup, the Cilium CNI plugin is called attach BPF programs to the
+network devices set up by aws-cni to enforce network policies, perform
+load-balancing, and encryption.
+
+.. image:: aws-cni-architecture.png
+
+
+Setup Cluster on AWS
+====================
+
+Follow the instructions in the :ref:`k8s_install_eks` guide to set up an EKS
+cluster or use any other method of your preference to set up a Kubernetes
+cluster.
+
+Ensure that the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_
+plugin is installed. If you have set up an EKS cluster, this is automatically
+done.
+
+Prepare Cilium to use AWS-CNI chaining
+======================================
+
+Download the Cilium deployment yaml:
+
+.. tabs::
+  .. group-tab:: K8s 1.14
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.14/cilium.yaml
+
+  .. group-tab:: K8s 1.13
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.13/cilium.yaml
+
+  .. group-tab:: K8s 1.12
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.12/cilium.yaml
+
+  .. group-tab:: K8s 1.11
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.11/cilium.yaml
+
+  .. group-tab:: K8s 1.10
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.10/cilium.yaml
+
+Edit ``cilium.yaml and add the following configuration to the ConfigMap:
+
+.. code:: bash
+
+      cni-chaining-mode: aws-cni
+      masquerade: "false"
+      tunnel: disabled
+
+This will enable chaining with the aws-cni plugin. It will also disable
+tunneling. Tunneling is not required as ENI IP addresses can be directly routed
+in your VPC. You can also disable masquerading for the same reason.
+
+Validate your Security Groups
+=============================
+
+Validate your AWS security groups rules and ensure that ENI IP addresses as
+allocated and used by the aws-cni plugin are allowed. See the documentation of
+the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_ plugin for
+more details.
+
+Deploy Cilium
+=============
+
+.. code:: bash
+
+       kubectl apply -f cilium.yaml
+
+As Cilium is deployed as a DaemonSet, it will write a new CNI configuration
+``05-cilium.conflist`` which will take precedence over the standard
+``10-aws.conflist``. Any new pod scheduled, will use the chaining configuration
+which will not also invoke Cilium.
+
+Restart existing pods
+=====================
+
+The new CNI chaining configuration will *not* apply to any pod that is already
+running the cluster. Existing pods will be reachable and Cilium will
+load-balance to them but policy enforcement will not apply to them and
+load-balancing is not performed for traffic originating from existing pods.
+You must restart these pods in order to invoke the
+chaining configuration on them.
+
```

Comment:
It is possible to know which pods are them, it's basically all pods that don't have a cilium endpoint in k8s and are not running in host network. We should make this as an opt-in in the operator for all pods and not only [kube|core]-dns.

Reply:
I don't like the operator restarting application pods. I don't think users will want that. The majority of users will install Cilium upfront before running app pods so the notice is fine I think. I've added a note to the docs to help users identify pods that need to be restarted.